core/__tests__/actions/destinations.ts

```diff
@@ -49,6 +49,7 @@ describe("actions/destinations", () => {
         name: "test destination",
         type: "test-plugin-export",
         appId: app.id,
+        syncMode: "sync",
       };
       const { error, destination } = await specHelper.runAction(
         "destination:create",
```

Comment:
You might want to also test that the data you get back in `destination:create` and `destination:view` include the sync mode.  This will in turn test that `Destination.apiData()` has the new syncMode included.